### PR TITLE
[TableGen] [NFC] Refine TableGen code to comply with `clang-tidy` checks

### PR DESCRIPTION
**Pull request by jerryyiransun**
Code cleanups for TableGen files, changes includes function names, variable names and unused imports.

### CODE REVIEW

File: llvm/utils/TableGen/CallingConvEmitter.cpp
Comment (jurahul) [inline llvm/utils/TableGen/CallingConvEmitter.cpp:37]: nit: can this be `OS`. I've seen that as commonly used variable name for raw_ostream?

**Comment by kparzysz** [comment]
The changes themselves are fine.  Such large-scale commits are usually avoided, but we've gone through various renaming efforts in the past.  I'm fine with committing this, but let's check with @Paul-C-Anagnostopoulos.

**Comment by jurahul** [comment]
What’s the best way to do this? Split it per file?

File: llvm/utils/TableGen/CallingConvEmitter.cpp
Comment (jerryyiransun) [reply llvm/utils/TableGen/CallingConvEmitter.cpp:37]: Do you want me to change all `&O` to `&OS` in the file? There are quite a few methods in this file that have raw_ostream as parameter and they are all defined to be `&O`

**Comment by redstar** [comment]
> What’s the best way to do this? Split it per file?

The removal of unused includes could be separated into a new PR. That would reduce the number of touched files, without creating a flood of PRs.

**Comment by kparzysz** [comment]
> What’s the best way to do this? Split it per file?

The issue is that "formatting" commits introduce noise in the file history, which can affect things like `git blame`.  There are ways around it, for example, blame allows you to give a list of commits to ignore (we have it in `.git-blame-ignore-revs`), but you still have to manually provide that option.  Some tools may not be doing it, and so they can trace some change back to an NFC commit.

The "preferred" approach it to tack it on top of some functional commit, i.e. make such changes to the code that the commit was changing anyway.  The downside is that this process may never finish all the intended changes.

Paul is the code owner of TableGen, so I defer to him to make the final decision.  This change is fine if there is a consensus to do it, and I'm not opposed.

**Comment by jurahul** [comment]
IMO, discouraging such changes to preserve git history is not a great
argument. We have tools like git blame which help recover the history. Yes
you may have to make a few more hops in the git blame chain. Also packaging
NFC and functional changes together is also not recommended AFAICT.

**Comment by Paul-C-Anagnostopoulos** [comment]
I'm not actively working on TableGen these days, so you don't really need my opinion. But these cleanups are a great idea and so go for it.

**Comment by jurahul** [comment]
Thanks. Though it might to get clarity on LLVM’s policy for such changes if
any. I’ll check if this has come up before and if there is any existing
guidance.

**Comment by jurahul** [comment]
https://discourse.llvm.org/t/widescale-clang-tidy-or-similar-based-cleanup/44527

Looks like wide scale code reformatting is discouraged but other cleanups
like this one are welcome.

For this case, if we can split into 2: unused header removal and rename
things to conform to LLVM coding standards seems reasonable.

**Comment by jurahul** [comment]
For the git blame issue, I wonder if git-blame can be taught to skip
commits with NFC keyword in the short description. On that topic, instead
of [code-cleanup] can you use [NFC]?

**Comment by jrtc27** [comment]
NFC doesn't mean ignorable. They can have a large amount of churn that's significant for blame, and they can end up not being NFC.

**Comment by tmatheson-arm** [comment]
These changes seem to be contrary to the [LLVM Coding Standards style recommendations](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly). Specifically variable and function names should start with a lower case letter.

File: llvm/utils/TableGen/CallingConvEmitter.cpp
Comment (jurahul) [reply llvm/utils/TableGen/CallingConvEmitter.cpp:37]: Yes, since you are changing the, you might as well use the preferred name

**Comment by jurahul** [comment]
Please also see https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly (@tmatheson-arm made a comment about this but I don't see it). Some of the changes are violating the LLVM's convention for names

**Comment by jerryyiransun** [comment]
Thanks for the feedback, I will review the webpage and make changes accordingly.

> Please also see https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly (@tmatheson-arm made a comment about this but I don't see it). Some of the changes are violating the LLVM's convention for names

**Comment by kparzysz** [comment]
> (@tmatheson-arm made a comment about this but I don't see it).

It was probably deleted by the author.

**Comment by tmatheson-arm** [comment]
> > (@tmatheson-arm made a comment about this but I don't see it).
> 
> It was probably deleted by the author.

Yes sorry, I did comment but then upon double checking it looked like I was wrong, so I deleted the comment to try and avoid confusion.

File: llvm/utils/TableGen/CallingConvEmitter.cpp
Comment (jerryyiransun) [reply llvm/utils/TableGen/CallingConvEmitter.cpp:37]: @jurahul should I create a new PR for this change? It looks like there are few other files that also needs this modification. I think it might introduce too many changed files in this PR.

**Comment by jerryyiransun** [comment]
I've extracted the commit for removing unused imports to #113725 as requested.

**Comment by jurahul** [comment]
Sounds good. Breaking it into smaller PR for independent changes is welcome.

**Comment by jurahul** [comment]
LGTM to me, but let's wait for other folks to chime in.

File: llvm/utils/TableGen/ARMTargetDefEmitter.cpp
Comment (arsenm) [inline llvm/utils/TableGen/ARMTargetDefEmitter.cpp:56]: Probably should keep ARM capitalized here

Comment (jerryyiransun) [reply llvm/utils/TableGen/ARMTargetDefEmitter.cpp]: fixed in most recent commit

**Review by arsenm — APPROVED** [review]

**Review by kparzysz — APPROVED** [review]

**Pull request merged**
